packages/dds/tree/src/feature-libraries/sequence-field/rebase.ts

```diff
@@ -89,9 +91,41 @@ function rebaseMarkList<TNodeChange>(
                 break;
             }
         } else if (isAttach(currMark)) {
-            if (isAttach(baseMark) && isAttachAfterBaseAttach(currMark, baseMark)) {
-                baseIter.pop();
-                factory.pushOffset(getOutputLength(baseMark));
+            if (isAttach(baseMark)) {
+                const reattachOffset = getOffsetInReattach(currMark.lineage, baseRevision);
+                if (reattachOffset !== undefined) {
+                    const offset = reattachOffset - baseReattachOffset;
+                    baseReattachOffset = reattachOffset;
+                    const reattachLength = getOutputLength(baseMark);
+
+                    if (offset >= reattachLength) {
+                        assert(
+                            offset === reattachLength,
+                            "Reattach is shorter than recorded detach",
+                        );
+                        baseIter.pop();
+                        factory.pushOffset(reattachLength);
+                    }
+                    if (offset > 0) {
+                        const [baseMarkBefore, baseMarkAfter] = splitMarkOnOutput(baseMark, offset);
+                        baseIter.push(baseMarkAfter);
```

Comment:
Assuming I was correct about the `else if` above, we need to pop `baseIter` before pushing the `baseMarkAfter` back in.